lib/cn/kvset.c

```diff
@@ -523,52 +487,28 @@ kvset_open2(
 
     assert(ks->ks_kvsetid != 0);
 
-    /* map single hblock */
-    err = mpool_mcache_mmap(mp, 1, &km->km_hblk_id, &ks->ks_hmap);
+    err = kvset_hblk_init(mp, km->km_hblk_id, &ks->ks_vgmap, &ks->ks_use_vgmap,
+        &ks->ks_hlog, &ks->ks_hblk);
     if (ev(err))
         goto err_exit;
 
-    {
-        struct mblock_props props;
-
-        err = mpool_mblock_props_get(mp, km->km_hblk_id, &props);
-        if (ev(err))
-            goto err_exit;
-
-        err = kvset_hblk_init(mp, &props, ks->ks_hmap, &ks->ks_vgmap, &ks->ks_use_vgmap,
-                              &ks->ks_hlog, &ks->ks_hblk);
-
-        if (ev(err))
-            goto err_exit;
-
-        /* kvset_stats from kblocks */
-        ks->ks_st.kst_halen += props.mpr_alloc_cap;
-        ks->ks_st.kst_hwlen += props.mpr_write_len;
-        ks->ks_st.kst_ptombs += ks->ks_hblk.kh_metrics.hm_nptombs;
-    }
+    /* kvset_stats from hblocks */
+    ks->ks_st.kst_halen += ks->ks_hblk.kh_hblk_desc.alen_pages * PAGE_SIZE;
+    ks->ks_st.kst_hwlen += ks->ks_hblk.kh_hblk_desc.wlen_pages * PAGE_SIZE;
+    ks->ks_st.kst_ptombs += ks->ks_hblk.kh_metrics.hm_nptombs;
 
     ks->ks_seqno_min = ks->ks_hblk.kh_seqno_min;
     ks->ks_seqno_max = ks->ks_hblk.kh_seqno_max;
     assert(ks->ks_seqno_min <= ks->ks_seqno_max);
 
-    /* map kblocks */
-    err = mpool_mcache_mmap(mp, km->km_kblk_list.idc, km->km_kblk_list.idv, &ks->ks_kmap);
-    if (ev(err))
-        goto err_exit;
-
     kcachesz = 0;
 
-    for (i = 0; i < n_kblks; i++) {
+    for (uint i = 0; i < n_kblks; i++) {
         struct kvset_kblk * kblk = ks->ks_kblks + i;
-        struct mblock_props props;
 
         u64 mbid = km->km_kblk_list.idv[i];
 
-        err = mpool_mblock_props_get(mp, mbid, &props);
-        if (ev(err))
-            goto err_exit;
-
-        err = kvset_kblk_init(rp, mp, &props, ks->ks_kmap, i, kblk);
+        err = kvset_kblk_init(rp, mp, mbid, kblk);
```

Comment:
We used to get props, then pass props into the read descriptor code.  Now we use `mblk_mmap` to get props and map, and then pass the map base into the read descriptors.  Enables much simplification in the readers.